tsMuxer/CMakeLists.txt

```diff
@@ -116,6 +116,76 @@ if(TSMUXER_STATIC_BUILD)
   endif()
 endif()
 
+# we can try compiling this small C++ program to test std::filesystem is working
+string(CONFIGURE [[
+  #include <filesystem>
+  int main( int argc, char ** argv ) {
+    std::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] STD_FS_CODE @ONLY)
+
+string(CONFIGURE [[
+  #include <experimental/filesystem>
+  int main( int argc, char ** argv ) {
+    std::experimental::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] EXP_FS_CODE @ONLY)
+
+string(CONFIGURE [[
+  #include <ghc/filesystem.hpp>
+  int main( int argc, char ** argv ) {
+    ghc::filesystem::path p( argv[ 0 ] );
+    return p.string().length();
+  }
+]] GHC_FS_CODE @ONLY)
```

Comment:
The `CONFIGURE` step isn't needed since there are no `@`-tokens in these strings. You can just use ordinary `set` to initialise these variables : 
```
set(STD_FS_CODE [[
  #include <filesystem>
  int main( int argc, char ** argv ) {
    std::filesystem::path p( argv[ 0 ] );
    return p.string().length();
  }
]])
```

Maybe it's not a big difference, but I find it significantly less confusing.

Using `string(CONFIGURE)` here would be possible if there was one "template" variable which was then "configured" for each variant of `fs` ... something like

```
set(TEMPLATE_FS_CODE [[
  #include <@HEADERFILE>
  using fs = @NAMESPACE;
  int main( int argc, char ** argv ) {
    fs::path p( argv[ 0 ] );
    return p.string().length();
  }
]])
```

Then you could call `string(CONFIGURE)` for each of the combinations to get the final source code to compile, setting `HEADERFILE` and `NAMESPACE` before each call. I like the explicit version much more, this feels like overcomplicating things for no good reason.